### ISSUE TITLE
Implement LinkedHashSet implementing MutableSet API.

### ISSUE THREAD

**Issue by donraab** [eclipse/eclipse-collections#339]

**Comment by oehme** [comment]
I think the title should be "implement insertion ordered set". Using a linked structure is the reason for the huge memory overhead of LinkedHashSet. Using an IntList to maintain the iteration order would be much more efficient. But maybe I'm missing a different reason why a linked structure might be useful?

**Comment by donraab** [comment]
Makes sense.  Updated.  I'll leave implementation detail up to the developer who contributes.

**Comment by motlin** [comment]
A removal from the middle of an IntList would turn Set.remove() into a
linear time operation, at least with the naive/obvious implementation. In
fact, any removal is complicated.

**Comment by oehme** [comment]
I corrected my comment above - I meant keeping the links in a long[], storing the previous pointer in the lower half and the next pointer in the upper half of each long and indexing it with the same indexes as the main table. That gives you a compact link representation and constant time removals.

**Comment by arvidvillen** [comment]
Any updates in this?
Surely `ImmutableOrderedSet` would be one of the most used collections in a regular project as the ordering should/could be done in DB query.

**Comment by nikhilnanivadekar** [comment]
@arvidvillen what updates would you like? If you need this functionality, please feel free to contribute.

**Comment by arvidvillen** [comment]
Thanks for super fast replies.
Interesting, I guess I'm not really thinking of it from a performance point of view, but rather the semantic meaning (what the interface tells the user about the collection). `ImmutableOrderedSet ` would tell me more as a consumer of the API than `ImmutableSet `or `ImmutableList`.

But I understand not many people see it that way as there are very few collection libraries that support `OrderedSet ` or `UniqueList `.
